### ISSUE TITLE
Create a Hinting system that will guide players towards using better weapons/classes 

### ISSUE THREAD

**Issue by jonsextra** [TheRealNin/Siege-Collection#53]
PASSED with a vote of 7/0/1 (y/n/a)

**Comment by TheRealNin** [comment]
Is this done using shine's adverts?

**Comment by DalinSeivewright** [comment]
I was thinking it could be implemented on the armory UI itself.
But if we think that's too much work there's no reason why it can't be some sort of text notification that appears at the bottom of the screen.  It'd have to stand out though.

I didn't know it was assigned to me either.

If we might be switching to your UI for Armory, it may be better to implement against that somehow.

**Comment by TheRealNin** [comment]
Ah! It could probably be integrated into my Armory GUI Overhaul. Maybe a simple "x2" the way aliens have on their buy screen for the number of marines with it equipped. Not sure if that data is actually available to the clients, since some of their team might not be in relevancy range.... hmm.

I think we need to get a better idea of what we want players to be picking from armories/protos. As we've found, no weapon is truly "bad".

**Comment by DalinSeivewright** [comment]
This is true but the idea is to steer the players towards weapons that will 100% benefit the team.  GLs and Railguns are sort of the niche weapon that you actually only want to be using in a few situations.  For the most part everything else is better, especially for newer players.

And the counts are fine but not enough on its own.  The counts themselves tells you what is in play but it doesn't communicate to the player (especially new players) what weapons/exo they should actually be choosing.  If they see everyone with MGs, they might decide a GL is what they should go for.  Also, if they see a bunch of people with GLs, they might just decide GLs are the thing to get and perpetuate the madness.  Then again, they could just up and ignore the counts (like I pretty much do for aliens).  I wish that description was added here but there was certain thresholds we worked out.  Basically, if 2/3 of the marine team wasn't a minigun exo it would indicate it is needed, otherwise it would go minigun/flamethrower or whatever  the ratios were (which are obviously tweakable).

*Edit:*
This is what Jon had worked out and I didn't think it was half bad:
`Marines: 2/3 exo (and no more than 1/4 of that railgun), 1/3 JP (of which 2/3 are Shotgun/mg/rifle, and 1/3 are flamethrowers) `

**Comment by jonsextra** [comment]
Originally suggested values:

Marines: 2/3 exo (and no more than 1/4 of that railgun), 1/3 JP (of which 2/3 are Shotgun/mg/rifle, and 1/3 are flamethrowers) 
Aliens: 2/3 onos. Of the remainder, 1/2 gorge, 1/4 lerk, 1/4 fade.

**Comment by TheRealNin** [comment]
How in the world would you enforce the number of exos and/or mgs? If you die and drop it, does that count as a mg? If you buy and exo and drop it in base, is that an exo?

**Comment by jonsextra** [comment]
It's a hinting system, no enforcement. All it would do is take a count of the current active classes, and then, using the armory/proto GUI and possibly chat hints, guide people towards the above values.

**Comment by Frozenclaw** [comment]
So kinda like the vanilla hints but have it deal with siege?

**Comment by jonsextra** [comment]
Not really. It'd be showing up more in the evolve menu and armory/proto gui.
